### PR TITLE
gr-osmosdr 0.2.4 (new formula)

### PR DESCRIPTION
**Pull request by tesch1**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/gr-osmosdr.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  depends_on "soapysdr"
+
+  def install
+    system "cmake", "-S", ".", "-B", "build", *std_cmake_args, "-DENABLE_NONFREE=TRUE"
+    system "cmake", "--build", "build"
+    system "cmake", "--install", "build"
+  end
+
+  test do
+    system "false"
```
Comment (chenrui333) [inline Formula/gr-osmosdr.rb:22]: Can you propose a more substantial test than checking the version or calling for help? These tests rarely ever fail even for broken installations.

>We want tests that don’t require any user input and test the basic functionality of the application. For example `foo build-foo input.foo` is a good test and (despite their widespread use) `foo --version` and `foo --help` are bad tests.

Ref. https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula

**Comment by chenrui333** [comment]
@tesch1 can you take a look at this PR? #93306

File: Formula/gr-osmosdr.rb (hunk 1 above)
Comment (tesch1) [reply Formula/gr-osmosdr.rb]: @chenrui333 I've added a little test to make sure the python imports. The issue is that the interface basically needs hardware to connect to.

**Comment by tesch1** [comment]
@rjpilgrim does this version work for you?

**Comment by tesch1** [comment]
fyi @Hoernchen - since you appear to be the primary maintainer of `gr-osmosdr` 👋

**Comment by darthcloud** [comment]
So I tested this new formula on my Mac Mini M2 pro and it install and work fine.
I was able to do the Great Scott Gadgets Gnuradio SDR tutorial with it.

This is the only package missing to be able to run this very common tutorial with a 100% vanilla brew install.
A lot of (now outdated) documentations refer to this formula, but otherwise the doc is still fine beside this missing formula.

Look like @tesch1 addressed all the comment in this thread.

I wonder if I can help in anyway to get this merged?

Thank you,
Jacques

**Pull request closed without merging**